### PR TITLE
Unification merge conflict

### PR DESCRIPTION
**Pull request by tgrego**
Do not merge!

This branch is a merge of master and feature/unified-design branches, that have diverged...
Apparently both branches are used in different places, so the merge has to be correct for master to become the only branch used.

Some minor fixes added.

### CODE REVIEW

**Comment by avullo** [comment]
I'm not sure what's to be reviewed here. To me, it looks like this is acceptable provided this branch works in all circumstances either master and unified-design are applied.

**Comment by nerdstrike** [comment]
Better get that JSON dependency into the cpanfile.

I think I missed this: https://github.com/Ensembl/ensembl-io/blob/970e00dd7d3c3cf3395c573a2f54acd78491cf14/travisci/harness.sh#L3

It should match the installed BioPerl 1.6.924.

File: modules/Bio/EnsEMBL/IO/Format/GFF3.pm [hunk 1]
```diff
@@ -0,0 +1,119 @@
+=pod
+
+=head1 LICENSE
+
+Copyright [1999-2015] Wellcome Trust Sanger Institute and the EMBL-European Bioinformatics Institute
+Copyright [2016] EMBL-European Bioinformatics Institute
```
Comment (nerdstrike) [inline modules/Bio/EnsEMBL/IO/Format/GFF3.pm]: Smoking gun of incorrect merge direction

File: modules/Bio/EnsEMBL/IO/Format/GTF.pm [hunk 2]
```diff
@@ -0,0 +1,119 @@
+=pod
+
+=head1 LICENSE
+
+Copyright [1999-2015] Wellcome Trust Sanger Institute and the EMBL-European Bioinformatics Institute
+Copyright [2016] EMBL-European Bioinformatics Institute
```
Comment (nerdstrike) [inline modules/Bio/EnsEMBL/IO/Format/GTF.pm]: Here too

File: modules/Bio/EnsEMBL/IO/Object/GFF3.pm [hunk 3]
```diff
@@ -0,0 +1,72 @@
+=head1 LICENSE
+
+Copyright [1999-2016] Wellcome Trust Sanger Institute and the EMBL-European Bioinformatics Institute
+
```
Comment (nerdstrike) [inline modules/Bio/EnsEMBL/IO/Object/GFF3.pm]: This one is even more retro.

File: modules/Bio/EnsEMBL/IO/Object/Metadata.pm [hunk 4]
```diff
@@ -0,0 +1,48 @@
+=head1 LICENSE
+
+Copyright [1999-2015] Wellcome Trust Sanger Institute and the EMBL-European Bioinformatics Institute
```
Comment (nerdstrike) [inline modules/Bio/EnsEMBL/IO/Object/Metadata.pm]: Each of these copyrights is older than the last!

File: modules/Bio/EnsEMBL/IO/Translator.pm [hunk 5]
```diff
@@ -15,53 +15,222 @@ WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
...
+}
+
+=head2 get_field
+
+    Description: Fetch a field from the object, use the callback set by
+                 derived classes to access the field from the object. If
+                 the callback doesn't exist, return undef
+    Args[1]    : Object to fetch fields from
+    Args[2]    : Field name
+    Returntype : String, hashref or undef
```
Comment (nerdstrike) [inline modules/Bio/EnsEMBL/IO/Translator.pm]: Ugh.

File: modules/Bio/EnsEMBL/IO/Translator/Feature.pm [hunk 6]
```diff
@@ -19,149 +17,539 @@ limitations under the License.
...
+    my %attrs;
+    my @ordered_keys = grep { exists $summary{$_} } qw(id Name Alias Parent Target Gap Derives_from Note Dbxref Ontology_term Is_circular);
+    my @ordered_values = @summary{@ordered_keys};
+    while (my $key = shift @ordered_keys) {
+	    my $value = shift @ordered_values;
+	    delete $summary{$key};
+	    if ($value && $value ne '') {
+	      if ($key =~ /id/) {
+          $key = uc($key);
+		      if ($object->isa('Bio::EnsEMBL::Transcript')) {
```
Comment (nerdstrike) [inline modules/Bio/EnsEMBL/IO/Translator/Feature.pm]: Mmmmm, I smell tabs

File: modules/Bio/EnsEMBL/IO/Translator/Feature.pm [hunk 7]
```diff
@@ -19,149 +17,539 @@ limitations under the License.
...
+    if( ! defined( $self->{'mapper'} ) ) {
+	    my $oa = Bio::EnsEMBL::Registry->get_adaptor('multi', 'ontology', 'OntologyTerm');
+	    $self->{'mapper'} = Bio::EnsEMBL::Utils::SequenceOntologyMapper->new($oa);
+    }
+
+    return $self->{'mapper'};
 }
 
-=head2 get_blockSizes
+=head2 _default_score
```
Comment (nerdstrike) [inline modules/Bio/EnsEMBL/IO/Translator/Feature.pm]: POD disagreement with sub name?

File: travisci/harness.sh [hunk 8]
```diff
@@ -1,18 +1,17 @@
 #!/bin/bash
 
-ENSDIR="${ENSDIR:-$PWD}"
+export PERL5LIB=$PWD/bioperl-live-bioperl-release-1-2-3:$PWD/ensembl-test/modules:$PWD/ensembl/modules:$PWD/modules:$PWD/ensembl-variation/modules:$DEPS/Bio-HTS/blib/lib/:$DEPS/Bio-HTS/blib/arch:$PERL5LIB
```
Comment (nerdstrike) [inline travisci/harness.sh]: This looks very wrong.

Comment (tgrego) [reply travisci/harness.sh]: indeed... was trying to fix it but gave up for now and reverted to that original state

**Comment by tgrego** [comment]
Finding it will be easier to feature complete before merge...
keeping this for reference.

**Pull request closed without merging**